From: msvensson at mysql dot com
Date: March 16 2006 11:39am
Subject: bk commit into 5.0 tree (msvensson:1.2086) BUG#15069

Below is the list of changes that have just been committed into a local
5.0 repository of msvensson. When msvensson does a push these changes will
be propagated to the main repository and, within 24 hours after the
push, to the public repository.
For information on how to access the public repository
see http://dev.mysql.com/doc/mysql/en/installing-source-tree.html

ChangeSet
  1.2086 06/03/16 11:39:41 msvensson@shellback.(none) +2 -0
  Bug#15069 [patch] my_print_defaults does'nt care --sysconfdir
   - Grab the path from "configure --sysconfdir=<path>" and set it as
     the first place to look for my.cnf files

  mysys/default.c
    1.81 06/03/16 11:39:33 msvensson@shellback.(none) +9 -1
    Use DEFAULT_SYSCONFDIR as first place to look for my.cnf file if "--sysconfdir=<path>" been specified as argument to configure

  mysys/Makefile.am
    1.68 06/03/16 11:39:33 msvensson@shellback.(none) +1 -0
    Grab the path from "--sysconfdir=<path>" into the defined variable DEFAULT_SYSYCONFDIR 

# This is a BitKeeper patch.  What follows are the unified diffs for the
# set of deltas contained in the patch.  The rest of the patch, the part
# that BitKeeper cares about, is below these diffs.
# User:	msvensson
# Host:	shellback.(none)
# Root:	/home/msvensson/mysql/bug15069/my50-bug15069

--- 1.67/mysys/Makefile.am	2005-10-14 01:13:55 +02:00
+++ 1.68/mysys/Makefile.am	2006-03-16 11:39:33 +01:00
@@ -70,6 +70,7 @@
 			-DSHAREDIR="\"$(MYSQLSHAREdir)\"" \
 			-DDEFAULT_HOME_ENV=MYSQL_HOME \
 			-DDEFAULT_GROUP_SUFFIX_ENV=MYSQL_GROUP_SUFFIX \
+                        -DDEFAULT_SYSCONFDIR="\"$(sysconfdir)\"" \
                         @DEFS@
 
 libmysys_a_DEPENDENCIES= @THREAD_LOBJECTS@

--- 1.80/mysys/default.c	2005-09-30 12:25:36 +02:00
+++ 1.81/mysys/default.c	2006-03-16 11:39:33 +01:00
@@ -48,7 +48,7 @@
 
 /* Which directories are searched for options (and in which order) */
 
-#define MAX_DEFAULT_DIRS 6
+#define MAX_DEFAULT_DIRS 7
 const char *default_directories[MAX_DEFAULT_DIRS + 1];
 
 #ifdef __WIN__
@@ -952,6 +952,10 @@
 /*
   Create the list of default directories.
 
+  If "--sysconfdir=<path>" was passed to configure, the
+  path will be added first in the list of default directories
+  on all platforms.
+
   On Microsoft Windows, this is:
     1. C:/
     2. GetWindowsDirectory()
@@ -983,6 +987,10 @@
 static void init_default_directories()
 {
   const char *env, **ptr= default_directories;
+
+#ifdef DEFAULT_SYSCONFDIR
+  *ptr++= DEFAULT_SYSCONFDIR;
+#endif
 
 #ifdef __WIN__
   *ptr++= "C:/";
